bundles/framework/publisher2/view/PanelGeneralInfo.js

```diff
@@ -1,178 +1,112 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
 import { GeneralInfoForm } from './form/GeneralInfoForm';
-import { ThemeProvider } from 'oskari-ui/util';
+import { PropTypes } from 'prop-types';
 
-/**
- * @class Oskari.mapframework.bundle.publisher2.view.PanelGeneralInfo
- *
- * Represents the basic info (name, domain, language) view for the publisher
- * as an Oskari.userinterface.component.AccordionPanel
- */
-Oskari.clazz.define('Oskari.mapframework.bundle.publisher2.view.PanelGeneralInfo',
-
-    /**
-     * @method create called automatically on construction
-     * @static
-     * @param {Object} sandbox
-     * @param {Object} localization
-     *       publisher localization data
-     */
-    function (sandbox, localization) {
-        this.loc = localization;
-        this.sandbox = sandbox;
+export const PANEL_GENERAL_INFO_ID = 'panelGeneralInfo';
+export class PanelGeneralInfo extends React.Component {
```

Comment:
We should try making these function components. Would that mean that the panel is passed some onChange function that is called when the user changes stuff? We can probably move the name/domain validation to some helper functions instead of having them on the component itself.

Reply:
Yeah. The first draft was somewhat quirky. Took a step back and separated the component + state handling in different places so now we have a function component paired with a separate statehandler/controller along with some additional logic to make it all work with publisher sidebar and antd-collapse. So far this applies only to generalinfopanel but I'll do map sizes panel next.